test/common/upstream/health_check_fuzz_test_utils.h

```diff
@@ -11,15 +11,18 @@
 
 namespace Envoy {
 namespace Upstream {
-
+// TODO IN THIS PR, IF YOU ARE GOING TO EDIT THESE TO NICEMOCKS, MAKE SURE TO CHECKOUT MASTER FOR
+// THESE UNIT TESTS BEFORE I EVER STEPPED FOOT HERE AT GOOGLE
```

Comment:
Please add a link to the exact commit and file.

Reply:
Whoops, sorry for checking this comment in. Basically Adi, for my health check fuzz test, at the beginning of the internship I pulled some logic/classes/state out of unit tests into this file. Thus, if I wrap around a NiceMock around objects here, it's tightly coupled with the unit tests, rendering EXPECT_CALLS unneeded. Thus, I took what was originally in master from before my internship, of course manually merged with new commits and logic tested, and moved all the logic back to health_checker_impl_test.cc.

Reply:
Deleted comment.

Reply:
Here's what was originally in the unit test classes (from master on my fork): https://github.com/zasweq/envoy/blob/master/test/common/upstream/health_checker_impl_test.cc